lib/dialects/postgres/query.js

```diff
@@ -131,6 +132,10 @@ Query.prototype.run = function(sql, parameters) {
       }
     }
 
+    if (self.options.ensureAffectedRows && result.rowCount < 1) {
+      throw new sequelizeErrors.NoAffectedRowsError('No rows affected by: ' + sql);
+    }
```

Comment:
My first move was to handle it on the legacy "Instance.save" (now would be Model).

But when I started doing this I noticed that each database handles it's metadata differenty, somethimes don't send nothing, names didn't match, and sometimes send it as data.

Then, the main two options that I considered were:

1. Changing Query.run signature responding something different when ensureAffectedRows is true, so I could handle at Model
2. Handing in a database-specific way

Then I choose 2.

What do you think? Another idea that I disconsidered?